test/core/regressions.js

```diff
@@ -185,4 +185,24 @@ describe("Core htmx Regression Tests", function(){
         btn.innerText.should.equal("FooBar");
     })
 
+
+    it("can trigger swaps from fields that don't support setSelectionRange", function(){
+        const template = '<form method="get" action="/test" hx-boost="true">\n' +
+            '<input value="test@test.com" type="email" id="id_email" />\n' +
+            '<input type="submit" id="id_button" />\n' +
+              '</form>';
+
+        const response = '<form>\n' +
+              '<input value="supertest@test.com" type="email" id="id_email" />\n' +
+              '<input type="submit" id="id_button" />\n' +
+              '</form>';
+        this.server.respondWith("GET", "/test", response);
+        make(template);
+        var input = byId("id_email");
+        input.focus();
+        input.dispatchEvent(new KeyboardEvent('keypress', {"key": "Enter"}));
```

Comment:
maybe make the trigger a click and then `click()` the input?  I don't think which event happens matters so much...